index.js

```diff
@@ -67,6 +87,27 @@ module.exports = function (src, dest, opts) {
 				resolve(true);
 			});
 
+			read.on('data', function () {
+				var written = write.bytesWritten;
+				var percent = (written / progress.stat.size).toPrecision(2);
+				opts.onProgress({
+					src: src,
+					dest: dest,
+					size: progress.stat.size,
+					written: written,
+					percent: percent
+				});
+			});
+
+			read.on('close', function () {
+				opts.onProgress({
```

Comment:
It should always return an object with the same properties. This object is different from the one in the `data` event.